restli-common/src/main/java/com/linkedin/restli/restspec/RestSpecCodec.java

```diff
@@ -81,8 +80,27 @@ public RestSpecCodec()
   public ResourceSchema readResourceSchema(InputStream inputStream)
           throws IOException
   {
+    return readResourceSchema(inputStream, false);
+  }
+
+  /**
+   * Reads a ResourceSchema from the given input stream, fixes it if necessary, and returns it.
+   *
+   * @param inputStream inputStream to read the ResourceSchema from
+   * @param generateLowercasePath if true and namespace exists in data map, then lower case the namespace value; Otherwise, no change
+   * @return a ResourceSchema
+   * @throws IOException
+   */
+  public ResourceSchema readResourceSchema(InputStream inputStream, boolean generateLowercasePath)
+          throws IOException{
     final DataMap data = _dataCodec.readMap(inputStream);
     fixupLegacyRestspec(data);
+    if (generateLowercasePath && data.containsKey("namespace")) {
+      String namespace = data.getString("namespace");
+      if (namespace != null) {
+        data.put("namespace", namespace.toLowerCase());
```

Comment:
Same here, avoid changing the namespace of the schema

Reply:
I would love to get on a call and discuss this with you. 

If i remove this, then the the paths will not change for the builders.